Can't create a project from existing sources with a non-downloadable owl:import

Input: an OWL ontology with an owl:import refering to a remote OWL ontology, but the URL is not dereferenciable/non-downloadable.

Steps to reproduce:

1. Press "Create New Project",
1. Choose the ontology file. Tested both with Turtle and XML serializations.
1. Press "Create New Project".

Expected behaviour: The project is created and the ontology is preloaded. Probably, a warning message saying that one of the owl:imports was not loaded...
Actual behaviour: A dialog window with title "Please referesh your browser"

Environment:

  - WebProtege 5.0.0:20200731:1910a39
  - Google Chrome 84.0.4147.105 (Official Build) (64-bit)

Logs:

```
2020-08-03 10:58:08.647  INFO   Received upload request from UserId{admin} at 10.183.157.151
--
  | 2020-08-03 10:58:08.761  INFO   File size is 59129 bytes.  Computed file size is 59129 bytes.
  | 2020-08-03 10:58:08.762  INFO   Stored uploaded file with name upload-615274650165342934
  | 2020-08-03 10:58:08.895  INFO   ProjectId{7e972803-b63f-40b7-bd7f-88247a7db26d} Creating project from sources
  | 2020-08-03 10:58:08.946  ERROR  An error occurred whilst executing an action
  | org.semanticweb.owlapi.model.UnloadableImportException: Could not load imported ontology: <http://data.internal.com/ontology/rdm/>  Cause: null
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.makeLoadImportRequest(OWLOntologyManagerImpl.java:1909)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.TripleHandlers$TPImportsHandler.handleTriple(TripleHandlers.java:1651)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.TripleHandlers$HandlerAccessor.handleStreaming(TripleHandlers.java:278)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer.statementWithResourceValue(OWLRDFConsumer.java:1558)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.RDFParser.statementWithResourceValue(RDFParser.java:370)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.EmptyPropertyElement.startElement(StartRDF.java:236)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.PropertyElementList.startElement(StartRDF.java:658)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.RDFParser.startElement(RDFParser.java:201)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:510)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
  | at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:351)
  | at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2710)
  | at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
  | at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
  | at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
  | at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
  | at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
  | at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:324)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.RDFParser.parse(RDFParser.java:145)
  | at org.semanticweb.owlapi.rdf.rdfxml.parser.RDFXMLParser.parse(RDFXMLParser.java:73)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyFactoryImpl.loadOWLOntology(OWLOntologyFactoryImpl.java:221)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.actualParse(OWLOntologyManagerImpl.java:1293)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.loadOntology(OWLOntologyManagerImpl.java:1233)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.loadOntologyFromOntologyDocument(OWLOntologyManagerImpl.java:1190)
  | at edu.stanford.bmir.protege.web.server.project.RawProjectSourcesImporter.importRawProjectSources(RawProjectSourcesImporter.java:31)
  | at edu.stanford.bmir.protege.web.server.upload.UploadedOntologiesProcessor.loadOntologies(UploadedOntologiesProcessor.java:55)
  | at edu.stanford.bmir.protege.web.server.upload.UploadedOntologiesProcessor.getUploadedOntologies(UploadedOntologiesProcessor.java:45)
  | at edu.stanford.bmir.protege.web.server.project.ProjectImporter.createProjectFromSources(ProjectImporter.java:68)
  | at edu.stanford.bmir.protege.web.server.project.ProjectCache.getProject(ProjectCache.java:206)
  | at edu.stanford.bmir.protege.web.server.project.ProjectManager.createNewProject(ProjectManager.java:68)
  | at edu.stanford.bmir.protege.web.server.project.CreateNewProjectActionHandler.execute(CreateNewProjectActionHandler.java:97)
  | at edu.stanford.bmir.protege.web.server.project.CreateNewProjectActionHandler.execute(CreateNewProjectActionHandler.java:41)
  | at edu.stanford.bmir.protege.web.server.dispatch.impl.DispatchServiceExecutorImpl.execAction(DispatchServiceExecutorImpl.java:136)
  | at edu.stanford.bmir.protege.web.server.dispatch.impl.DispatchServiceExecutorImpl.execute(DispatchServiceExecutorImpl.java:89)
  | at edu.stanford.bmir.protege.web.server.dispatch.DispatchServlet.executeAction(DispatchServlet.java:57)
  | at jdk.internal.reflect.GeneratedMethodAccessor26.invoke(Unknown Source)
  | at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  | at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  | at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
  | at edu.stanford.bmir.protege.web.server.app.WebProtegeRemoteServiceServlet.processCall(WebProtegeRemoteServiceServlet.java:64)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
  | at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at edu.stanford.bmir.protege.web.server.filter.WebProtegeWebAppFilter.doFilter(WebProtegeWebAppFilter.java:75)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
  | at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
  | at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
  | at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
  | at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
  | at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:688)
  | at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
  | at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
  | at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:609)
  | at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
  | at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
  | at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1623)
  | at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
  | at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  | at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  | at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
  | at java.base/java.lang.Thread.run(Thread.java:834)
  | Caused by: org.semanticweb.owlapi.io.OWLOntologyCreationIOException: null
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyFactoryImpl.loadOWLOntology(OWLOntologyFactoryImpl.java:231)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.actualParse(OWLOntologyManagerImpl.java:1293)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.loadOntology(OWLOntologyManagerImpl.java:1233)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.loadOntology(OWLOntologyManagerImpl.java:1133)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.loadImports(OWLOntologyManagerImpl.java:1864)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyManagerImpl.makeLoadImportRequest(OWLOntologyManagerImpl.java:1902)
  | ... 73 common frames omitted
  | Caused by: java.io.EOFException: null
  | at java.base/java.io.DataInputStream.readInt(DataInputStream.java:397)
  | at org.semanticweb.binaryowl.BinaryOWLOntologyDocumentPreamble.read(BinaryOWLOntologyDocumentPreamble.java:77)
  | at org.semanticweb.binaryowl.BinaryOWLOntologyDocumentPreamble.<init>(BinaryOWLOntologyDocumentPreamble.java:69)
  | at org.semanticweb.binaryowl.BinaryOWLOntologyDocumentSerializer.read(BinaryOWLOntologyDocumentSerializer.java:83)
  | at org.semanticweb.binaryowl.owlapi.BinaryOWLOntologyDocumentParser.parse(BinaryOWLOntologyDocumentParser.java:107)
  | at org.semanticweb.binaryowl.owlapi.BinaryOWLOntologyDocumentParser.parse(BinaryOWLOntologyDocumentParser.java:98)
  | at uk.ac.manchester.cs.owl.owlapi.OWLOntologyFactoryImpl.loadOWLOntology(OWLOntologyFactoryImpl.java:221)
  | ... 78 common frames omitted
  | 2020-08-03 10:58:08.951  ERROR  An unexpected exception was thrown on the server
  |  
  | Date and time: Mon Aug 03 10:58:08 UTC 2020
  |  
  | User: admin
  | Request URI: /webprotege/dispatchservice
  | Remote address: 10.251.16.1
  | Remote host: 10.251.16.1
  |  
  | Headers:
  | content-length: 272
  | x-gwt-module-base: https://webprotege-edsplat01-dev.inh-devapps.eu.internal.com/webprotege/
  | x-gwt-permutation: 9CD27B36499511B52BC9CCDFA2558FE2
  | user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36
  | content-type: text/x-gwt-rpc; charset=UTF-8
  | accept: */*
  | origin: https://webprotege-edsplat01-dev.inh-devapps.eu.internal.com
  | sec-fetch-site: same-origin
  | sec-fetch-mode: cors
  | sec-fetch-dest: empty
  | referer: https://webprotege-edsplat01-dev.inh-devapps.eu.internal.com/
  | accept-encoding: gzip, deflate, br
  | accept-language: en-US,en;q=0.9
  | cookie: crowd.token_key=hPVfnijBJ43F8CUlwuIigAAAAAAAYQABbWFrc2ltLmtvbGNoaW5AYm9laHJpbmdlci1pbmdlbGhlaW0uY29t; JSESSIONID=E1CD16FE3B5D92FDA12B2426CE2653E7
  | host: webprotege-edsplat01-dev.inh-devapps.eu.internal.com
  | x-forwarded-host: webprotege-edsplat01-dev.inh-devapps.eu.internal.com
  | x-forwarded-port: 443
  | x-forwarded-proto: https
  | forwarded: for=10.183.157.151;host=webprotege-edsplat01-dev.inh-devapps.eu.internal.com;proto=https;proto-version=
  | x-forwarded-for: 10.183.157.151
  |  
  |  
  | Message: could not get type signature for class org.semanticweb.owlapi.model.UnloadableImportException
  |  
  | Stack trace:
  |  
  | com.google.gwt.user.client.rpc.SerializationException: could not get type signature for class org.semanticweb.owlapi.model.UnloadableImportException
  | at com.google.gwt.user.client.rpc.impl.AbstractSerializationStreamWriter.writeObject(AbstractSerializationStreamWriter.java:126)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter$ValueWriter$8.write(ServerSerializationStreamWriter.java:167)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter.serializeValue(ServerSerializationStreamWriter.java:606)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter.serializeClass(ServerSerializationStreamWriter.java:787)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter.serializeImpl(ServerSerializationStreamWriter.java:826)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter.serialize(ServerSerializationStreamWriter.java:698)
  | at com.google.gwt.user.client.rpc.impl.AbstractSerializationStreamWriter.writeObject(AbstractSerializationStreamWriter.java:130)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter$ValueWriter$8.write(ServerSerializationStreamWriter.java:167)
  | at com.google.gwt.user.server.rpc.impl.ServerSerializationStreamWriter.serializeValue(ServerSerializationStreamWriter.java:606)
  | at com.google.gwt.user.server.rpc.RPC.encodeResponse(RPC.java:644)
  | at com.google.gwt.user.server.rpc.RPC.encodeResponseForFailure(RPC.java:419)
  | at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:605)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
  | at edu.stanford.bmir.protege.web.server.app.WebProtegeRemoteServiceServlet.processCall(WebProtegeRemoteServiceServlet.java:64)
  | at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
  | at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
  | at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at edu.stanford.bmir.protege.web.server.filter.WebProtegeWebAppFilter.doFilter(WebProtegeWebAppFilter.java:75)
  | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
  | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
  | at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
  | at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
  | at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
  | at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
  | at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
  | at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:688)
  | at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
  | at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
  | at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:609)
  | at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
  | at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
  | at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1623)
  | at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
  | at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  | at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  | at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
  | at java.base/java.lang.Thread.run(Thread.java:834)


```

--- a/webprotege-client/src/main/java/edu/stanford/bmir/protege/web/client/Messages.java
+++ b/webprotege-client/src/main/java/edu/stanford/bmir/protege/web/client/Messages.java
@@ -723,6 +723,10 @@ public interface Messages extends com.google.gwt.i18n.client.Messages {
     @Key("projectSettings.language.helpText")
     SafeHtml projectSettings_language_helpText();
 
+    @DefaultMessage("Any owl:import that cannot be loaded will be silently ignored")
+    @Key("projectSettings.owlimport.helpText")
+    SafeHtml projectSettings_owl_import_helpText();
+
     
     @DefaultMessage("Display name")
     @Key("projectSettings.displayName")

--- a/webprotege-client/src/main/java/edu/stanford/bmir/protege/web/client/project/CreateNewProjectViewImpl.ui.xml
+++ b/webprotege-client/src/main/java/edu/stanford/bmir/protege/web/client/project/CreateNewProjectViewImpl.ui.xml
@@ -38,6 +38,7 @@
                     <g:FileUpload ui:field="fileUpload"/>
                 </g:FormPanel>
             </g:HTMLPanel>
+            <label class="{wp.style.formHelpText} {style.helpText}"><ui:safehtml from="{msg.projectSettings_owl_import_helpText}"/></label>
         </g:HTMLPanel>
 
     </g:HTMLPanel>

--- a/webprotege-server-core/src/main/java/edu/stanford/bmir/protege/web/server/upload/UploadedOntologiesProcessor.java
+++ b/webprotege-server-core/src/main/java/edu/stanford/bmir/protege/web/server/upload/UploadedOntologiesProcessor.java
@@ -5,6 +5,7 @@ import edu.stanford.bmir.protege.web.server.project.Ontology;
 import edu.stanford.bmir.protege.web.server.project.RawProjectSourcesImporter;
 import edu.stanford.bmir.protege.web.server.project.UploadedProjectSourcesExtractor;
 import edu.stanford.bmir.protege.web.shared.csv.DocumentId;
+import org.semanticweb.owlapi.model.MissingImportHandlingStrategy;
 import org.semanticweb.owlapi.model.OWLOntology;
 import org.semanticweb.owlapi.model.OWLOntologyCreationException;
 import org.semanticweb.owlapi.model.OWLOntologyLoaderConfiguration;
@@ -50,7 +51,9 @@ public class UploadedOntologiesProcessor {
         var uploadedFile = documentResolver.resolve(documentId).toFile();
         var uploadedProjectSourcesExtractor = uploadedProjectSourcesExtractorProvider.get();
         var rawProjectSources = uploadedProjectSourcesExtractor.extractProjectSources(uploadedFile);
-        var loaderConfig = new OWLOntologyLoaderConfiguration();
+        var loaderConfig = new OWLOntologyLoaderConfiguration()
+                // See https://github.com/protegeproject/webprotege/issues/700
+                .setMissingImportHandlingStrategy(MissingImportHandlingStrategy.SILENT);
         var rawProjectSourcesImporter = new RawProjectSourcesImporter(manager, loaderConfig);
         rawProjectSourcesImporter.importRawProjectSources(rawProjectSources);
         return manager.getOntologies()

